### PR TITLE
[VBLOCKS-1910] Platform tokens

### PR DESCRIPTION
**Pull request by mhuynh5757**
## Submission Checklist

 - [ ] The `CHANGELOG.md` reflects any **feature**, **bug fixes**, or **known issues** made in the source code
 - [x] New unit tests and integration tests have beed added
 - [x] Code coverage is more or equal to the previous coverage
 - [x] A visual inspection of the `Files changed` tab was made prior to submitting the pull request ensuring the style guide was followed
 - [ ] [CI pipeline](https://app.circleci.com/pipelines/github/twilio/twilio-voice-react-native-app) is green
 - [x] ~Included screenshot if necessary~

## Description

This PR fixes the token vending mechanism to recognize the platform the token is being vended for, and adjust the incomingAllow and pushCredentialSid properties appropriately.

### Breakdown

- Include platform in client-side fetch.
- Parse platform on server.
- Tests for client and server.

## Validation

- Tests and manual validation on emulator.

## Additional Notes

N/A

## Contributing to Twilio

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

**Review by kpchoy — APPROVED** [review]

File: app/e2e/incomingCall.test.ts [hunk 1]
```diff
@@ -27,16 +27,18 @@ describe('Incoming Call', () => {
     await expect(element(by.text('Ahoy!'))).toBeVisible();
     await expect(element(by.text('test_email@twilio.com'))).toBeVisible();
 
-    await twilioClient.calls.create({
+    /**
+     * Wait for 10 seconds to let the registration settle.
+     */
+    await new Promise((resolve) => setTimeout(resolve, 10000));
```
Comment (charliesantos) [inline app/e2e/incomingCall.test.ts:33]: always 10s? How did we come up with this value?

File: app/e2e/incomingCall.test.ts [hunk 2]
```diff
@@ -51,6 +53,10 @@ describe('Incoming Call', () => {
     await device.reloadReactNative();
   });
 
+  afterEach(async () => {
+    await new Promise((resolve) => setTimeout(resolve, 10000));
```
Comment (charliesantos) [inline app/e2e/incomingCall.test.ts:57]: What is this for?

Comment (charliesantos) [reply app/e2e/incomingCall.test.ts:57]: Can we add a comment to this line so we don't forget?

File: app/e2e/incomingCall.test.ts [hunk 3]
```diff
@@ -27,16 +27,18 @@ describe('Incoming Call', () => {
...
       twiml: '<Response><Say>Ahoy, world!</Say><Pause length="5" /></Response>',
       to: `client:${clientId}`,
       from: 'detox',
     });
 
-    /**
-     * Wait for 10 seconds to let the registration settle.
-     */
-    await new Promise((resolve) => setTimeout(resolve, 10000));
+    console.log(`Call created with SID: "${testCall.sid}".`);
```
Comment (kpchoy) [inline app/e2e/incomingCall.test.ts:41]: Is this log necessary?

File: app/e2e/incomingCall.test.ts (hunk 2 above)
Comment (mhuynh5757) [reply app/e2e/incomingCall.test.ts:57]: Sounds good. Hopefully it doesn't flake again 😛

File: app/e2e/incomingCall.test.ts (hunk 1 above)
Comment (kpchoy) [reply app/e2e/incomingCall.test.ts:33]: 10s does feel like a long time to let the registration settle. Doesn't registration settle when you verify "Ahoy!" on line 27-28

File: app/e2e/incomingCall.test.ts (hunk 3 above)
Comment (mhuynh5757) [reply app/e2e/incomingCall.test.ts:44]: Helps with debugging if the call doesn't make it through somehow. Do we want to remove it? WDYT @charliesantos

File: app/e2e/incomingCall.test.ts (hunk 1 above)
Comment (mhuynh5757) [reply app/e2e/incomingCall.test.ts:36]: Yeah that's when the RN Promise is resolved by the native layer but I suspect there's some things that are happening on the native layer after that. It seems to help.

File: app/e2e/incomingCall.test.ts (hunk 3 above)
Comment (charliesantos) [reply app/e2e/incomingCall.test.ts:44]: I think it helps to know the callsid.

**Review by kpchoy — APPROVED** [review]

**Review by charliesantos — APPROVED** [review]

**Pull request merged**